addons/web/static/tests/public/interaction_service.test.js

```diff
@@ -0,0 +1,335 @@
+            expect.step("start notboom");
+        }
+    }
+
+    const { core } = await startInteraction([Boom, NotBoom], `<div class="test"></div>`, {
+        waitForStart: false,
+    });
+
+    let e = null;
+    try {
+        await core.isReady;
+    } catch (_e) {
+        e = _e;
+    }
+    expect(e.message).toBe(
+        "Could not start interaction Boom (invalid selector: 'div:invalid(coucou)')"
+    );
+
+    expect.verifySteps(["start notboom"]);
+});
+
+test("start interactions even if there is a crash when evaluating selectorHas", async () => {
+    class Boom extends Interaction {
+        static selector = ".test";
+        static selectorHas = "div:invalid(coucou)";
+
+        setup() {
+            expect.step("start boom");
+        }
+        destroy() {
+            expect.step("destroy boom");
+        }
+    }
+    class NotBoom extends Interaction {
+        static selector = ".test";
+
+        setup() {
+            expect.step("start notboom");
+        }
+    }
+
+    const { core } = await startInteraction(
+        [Boom, NotBoom],
+        `<div class="test"><div></div></div>`,
+        {
+            waitForStart: false,
+        }
+    );
+
+    let e = null;
+    try {
+        await core.isReady;
+    } catch (_e) {
+        e = _e;
+    }
+    expect(e.message).toBe(
+        "Could not start interaction Boom (invalid selector: '.test' or selectorHas: 'div:invalid(coucou)')"
+    );
+
+    expect.verifySteps(["start notboom"]);
+});
+
+test("start interactions with selectorHas", async () => {
+    class Test extends Interaction {
+        static selector = ".test";
+        static selectorHas = ".inner";
+
+        start() {
+            expect.step("start");
+        }
+    }
+
+    const { core } = await startInteraction(
+        Test,
+        `
+        <div class="test"><div class="inner"></div></div>
+        <div class="test"><div class="not-inner"></div></div>
+    `
+    );
+    expect(core.interactions).toHaveLength(1);
+    expect.verifySteps(["start"]);
+    expect(core.interactions[0].interaction.el).toBe(queryOne(".test:has(.inner)"));
+});
+
+test("recover from error as much as possible when applying dynamiccontent", async () => {
+    let a = "a";
+    let b = "b";
+    let c = "c";
+    let interaction = null;
+
+    class Test extends Interaction {
+        static selector = ".test";
+        dynamicContent = {
+            _root: {
+                "t-att-a": () => a,
+                "t-att-b": () => {
+                    if (b === "boom") {
+                        throw new Error("boom");
+                    }
+                    return b;
+                },
+                "t-att-c": () => c,
+            },
+        };
+        setup() {
+            interaction = this;
+        }
+    }
+
+    const { el } = await startInteraction(Test, `<div class="test"></div>`);
+
+    expect(el.querySelector(".test").outerHTML).toBe(`<div class="test" a="a" b="b" c="c"></div>`);
```

Comment:
Example here, you probably never need `querySelector`/`querySelectorAll` in tests (it is not discouraged, but I'm just pointing out that there is often a shorter syntax that will also offer more informations in test results):
```suggestion
    expect(".test:first").toHaveOuterHTML(`<div class="test" a="a" b="b" c="c"></div>`);
```